src/interface/viewport.c

```diff
@@ -1100,6 +1100,68 @@ int sub_98197C(
 
 /**
  *
+ *  rct2: 0x006D5DA9
+ */
+static void vehicle_visual_roto_drop(int x, int imageDirection, int y, int z, rct_vehicle *vehicle, int rct2VehiclePtrFormat)
+{
+	//RCT2_CALLPROC_X(0x006D5DA9, x, imageDirection, y, z, (int)vehicle, rct2VehiclePtrFormat, 0); return;
+	const rct_ride_entry_vehicle *vehicleEntry = (const rct_ride_entry_vehicle *)(rct2VehiclePtrFormat + 0x1A);
+	int image_id = (vehicle->var_C5 / 4) & 0x3;
+	image_id += vehicleEntry->base_image_id;
+	image_id += 4;
+	if (vehicle->restraints_position >= 64) {
+		image_id += 7;
+		image_id += (vehicle->restraints_position / 64);
+	}
+	int ebx;
+	ebx = image_id | (vehicle->colours.body_colour << 19 | vehicle->colours.trim_colour << 24 | 0xA0000000);
```

Comment:
can you group this with more brackets just to make it clearer. Also suggest changing `image_id` to `baseImage_id` and using `image_id` instead of your `ebx` variable.